src/main/java/com/hotels/beeju/extensions/HiveMetaStoreJUnitExtension.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+
+import com.hotels.beeju.core.HiveMetaStoreCore;
+
+/**
+ * A JUnit {@link Extension} that creates a Hive Metastore backed by an HSQLDB in-memory database.
+ * <p>
+ * A fresh database instance will be created for each test method.
+ * </p>
+ * To allow querying of the memory database, add the following to your {@code @BeforeEach} method. This will open a Swing
+ * SQL query window for the duration of the test. Remember to add a break point to your test.
+ *
+ * <pre>
+ * &#064;Rule
+ * public HiveMetaStoreJUnitRule hive = new HiveMetaStoreJUnitRule();
+ *
+ * &#064;Override
+ * protected void before() throws Throwable {
+ *   org.hsqldb.util.DatabaseManagerSwing
+ *       .main(new String[] {
+ *           &quot;--url&quot;,
+ *           hive.connectionURL(),
+ *           &quot;--user&quot;,
+ *           HiveMetaStoreJUnitRule.HSQLDB_USER,
```

Comment:
could you test if this still works, I think we are using Derby DB so those USER/PASSWORD things are wrong. We definitely shouldn't refer to the Rule here.